tests/sqlparser_common.rs

```diff
@@ -1228,16 +1228,33 @@ fn parse_string_agg() {
     );
 }
 
+macro_rules! dialect_is {
+    ( $dialect: ident is $($dialect_type: ty)|+ ) => {
+        ($($dialect.is::<$dialect_type>())||+)
+    };
+}
+
+/// selects all dialects but PostgreSQL
+pub fn all_dialects_but_pg() -> TestedDialects {
+    TestedDialects {
+        dialects: all_dialects()
+            .dialects
+            .into_iter()
+            .filter(|x| !dialect_is!(x is PostgreSqlDialect))
```

Comment:
since this macro (and frankly, function) are only used in one place, I'd prefer we just use the `.is::<PostgreSqlDialect>()` method directly